core_lib/src/structure/object.cpp

```diff
@@ -164,11 +164,8 @@ void Object::createWorkingDir()
         QFileInfo fileInfo(mFilePath);
         strFolderName = fileInfo.completeBaseName();
     }
-    const QString strWorkingDir = QDir::tempPath()
-        + "/Pencil2D/"
-        + strFolderName
-        + PFF_TMP_DECOMPRESS_EXT
-        + "/";
+    const QString strWorkingDir = 
```

Comment:
For my peace of mind, could we change the following few lines to this:
```
QString strWorkingDir;
QDir dir(QDir::tempPath());
do {
    strWorkingDir =
        QString("%1/Pencil2D/%2_%3_%4/").arg(QDir::tempPath()).arg(strFolderName).arg(PFF_TMP_DECOMPRESS_EXT).arg(uniqueString(8));
} while(dir.exists(strWorkingDir));
```
The chances of generating the same string is almost impossible, but I would feel better if we checked for it anyway.